src/ray/raylet/main.cc

```diff
@@ -83,8 +81,11 @@ int main(int argc, char *argv[]) {
   // Destroy the Raylet on a SIGTERM. The pointer to main_service is
   // guaranteed to be valid since this function will run the event loop
   // instead of returning immediately.
-  auto handler = [&main_service](const boost::system::error_code &error,
-                                 int signal_number) { main_service.stop(); };
+  auto handler = [&main_service, &raylet_socket_name](const boost::system::error_code &error,
+                                 int signal_number) {
+    main_service.stop();
+    remove(raylet_socket_name.c_str());
```

Comment:
What does this line do?

Reply:
When raylet process exit safely, the local socket file like `raylet11111` is always kept there.

Reply:
I see. Could you add a comment explaining that?

Reply:
My job.